src/AppInstallerCLICore/Workflows/UninstallFlow.cpp

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "UninstallFlow.h"
+#include "WorkflowBase.h"
+#include "MSStoreInstallerHandler.h"
+#include "ShellExecuteInstallerHandler.h"
+#include "AppInstallerMsixInfo.h"
+
+using namespace AppInstaller::Manifest;
+using namespace AppInstaller::Msix;
+using namespace AppInstaller::Repository;
+
+namespace AppInstaller::CLI::Workflow
+{
+    namespace
+    {
+        std::optional<DWORD> InvokeCreateProcess(const std::string& command, IProgressCallback& progress)
+        {
+            std::wstring commandUtf16 = Utility::ConvertToUTF16(command);
+
+            // Parse the command string as application and command line for CreateProcess
+            PWSTR app, commandLine;
+            THROW_IF_FAILED(SHEvaluateSystemCommandTemplate(commandUtf16.c_str(), &app, &commandLine, NULL));
```

Comment:
>SHEvaluateSystemCommandTemplate [](start = 28, length = 31)

I thought you explicitly wanted to use CreateProcess because it didn't require this.

Reply:
That was originally because I thought it would be harder to parse the string before I knew of this function. I ended using CreateProcess anyway because that's what the "Programs & Features" UI does.

Reply:
Chatted offline, we found "Programs & Features" uses CreateProcess and Settings(which is modern) uses ShellExecute.

I think for code consistency and extra ShellExecute benefits like ensuring policy, etc, It might be better to use ShellExecute here if not much refactoring effort needed?

---
In reply to: [535774167](https://github.com/microsoft/winget-cli/pull/659#discussion_r535774167) [](ancestors = 535774167)

Reply:
I think it makes sense to change it to ShellExecute. I'll do that.